src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -260,18 +263,79 @@ private void resetTuning() {
     numServesBlocking = 0;
   }
 
+  private static final Pattern RANGE = Pattern.compile("^bytes=(\\d+)-(\\d*)$");
+
   private void sendFile(RoutingContext context, String file, FileProps fileProps) {
     HttpServerRequest request = context.request();
+
+    Long offset = null;
+    Long end = null;
+    MultiMap headers = null;
+
+    if (rangeSupport) {
+      // check if the client is making a range request
+      String range = request.getHeader("Range");
+      // end byte is length - 1
+      end = fileProps.size() - 1;
+
+      if (range != null) {
+        Matcher m = RANGE.matcher(range);
+        if (m.matches()) {
+          try {
+            String part = m.group(1);
+            // offset cannot be empty
+            offset = Long.parseLong(part);
+            // offset must fall inside the limits of the file
+            if (offset < 0 || offset >= fileProps.size()) {
+              throw new IndexOutOfBoundsException();
+            }
+            // length can be empty
+            part = m.group(2);
+            if (part != null && part.length() > 0) {
+              // ranges are inclusive
+              end = Long.parseLong(part);
+              // offset must fall inside the limits of the file
+              if (end < offset || end >= fileProps.size()) {
+                throw new IndexOutOfBoundsException();
+              }
+            }
+          } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            context.fail(416);
```

Comment:
humm, all other http status are expressed by their numeral, 200, 403, 401, 206... why should the range not satisfiable be a constant? Or are you suggesting to replace all numerals with constants?